docker/oap-es7/docker-entrypoint.sh

```diff
@@ -294,202 +33,51 @@ validateVariables() {
-    sampleRate: \${SW_TRACE_SAMPLE_RATE:10000} # The sample rate precision is 1/10000. 10000 means 100% sample in default.
-    slowDBAccessThreshold: \${SW_SLOW_DB_THRESHOLD:default:200,mongodb:100} # The slow database access thresholds. Unit ms.
-receiver-jvm:
-  default:
-receiver-clr:
-  default:
-service-mesh:
-  default:
-    bufferPath: \${SW_SERVICE_MESH_BUFFER_PATH:../mesh-buffer/}  # Path to trace buffer files, suggest to use absolute path
-    bufferOffsetMaxFileSize: \${SW_SERVICE_MESH_OFFSET_MAX_FILE_SIZE:100} # Unit is MB
-    bufferDataMaxFileSize: \${SW_SERVICE_MESH_BUFFER_DATA_MAX_FILE_SIZE:500} # Unit is MB
-    bufferFileCleanWhenRestart: \${SW_SERVICE_MESH_BUFFER_FILE_CLEAN_WHEN_RESTART:false}
-istio-telemetry:
-  default:
-query:
-  graphql:
-    path: \${SW_QUERY_GRAPHQL_PATH:/graphql}
-alarm:
-  default:
-EOT
-    # generate telemetry
-    case ${SW_TELEMETRY} in
-    none) generateTelemetryNone;;
-    prometheus) generateTelemetryPrometheus;;
-    so11y) generateTelemetrySo11y;;
-    esac
+validateVariables "SW_CONFIGURATION" "$SW_CONFIGURATION" "none apollo nacos zookeeper"
 
-    # generate configuration
-    case ${SW_CONFIGURATION} in
-    none) generateConfigurationNone;;
-    apollo) generateConfigurationApollo;;
-    nacos) generateConfigurationNacos;;
-    zookeeper) generateConfigurationZookeeper;;
-    consul) generateConfigurationConsul;;
-    grpc) generateConfigurationGRPC;;
-    esac
+validateVariables "SW_TELEMETRY" "$SW_TELEMETRY" "none prometheus so11y"
 
-    cat <<EOT >> ${var_application_file}
-envoy-metric:
-  default:
-EOT
-    if [[ "$SW_ENVOY_ALS_ENABLED" = "true" ]]; then
-        cat <<EOT >> ${var_application_file}
-    alsHTTPAnalysis: \${SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS:k8s-mesh}
-EOT
-    fi
-
-    if [[ "$SW_RECEIVER_ZIPKIN_ENABLED" = "true" ]]; then
-        cat <<EOT >> ${var_application_file}
-receiver_zipkin:
-  default:
-    host: \${SW_RECEIVER_ZIPKIN_HOST:0.0.0.0}
-    port: \${SW_RECEIVER_ZIPKIN_PORT:9411}
-    contextPath: \${SW_RECEIVER_ZIPKIN_CONTEXT_PATH:/}
-EOT
-    fi
+if [[ "$SW_ENVOY_ALS_ENABLED" = "true" ]]; then
+    export SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS=k8s-mesh
+    echo "Set SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS to ${SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS}"
+fi
 
-    if [[ "$SW_RECEIVER_JAEGER_ENABLED" = "true" ]]; then
-        cat <<EOT >> ${var_application_file}
-receiver_jaeger:
-  default:
-    gRPCHost: \${SW_RECEIVER_JAEGER_HOST:0.0.0.0}
-    gRPCPort: \${SW_RECEIVER_JAEGER_PORT:14250}
-EOT
-    fi
+if [[ "$SW_RECEIVER_ZIPKIN_ENABLED" = "true" ]]; then
+    export SW_RECEIVER_ZIPKIN=default
+    echo "Set SW_RECEIVER_ZIPKIN to ${SW_RECEIVER_ZIPKIN}"
+fi
 
-    if [[ "$SW_TELEMETRY" = "so11y" ]]; then
-        cat <<EOT >> ${var_application_file}
-receiver-so11y:
-  default:
-EOT
-    fi
+if [[ "$SW_RECEIVER_JAEGER_ENABLED" = "true" ]]; then
+    export SW_RECEIVER_JAEGER=default
+    echo "Set SW_RECEIVER_JAEGER to ${SW_RECEIVER_JAEGER}"
+fi
```

Comment:
same

Reply:
Receiver enable seems a little different, @hanahmily. They are separated new module, not provider. We may need another feature to resolve this for docker.